Update wallet.md (#1174)
MystenLabs/sui: doc/src/build/wallet.md

1. Optionally, set `RUST_LOG=debug` for verbose logging.
1. Initiate `genesis`:
   ```shell
   $ sui genesis
   ```

All of this is contained in configuration and keystore files, as well
as an `authorities_db` database file.

> TODO: Why do I no longer see a `client_db` file created, only `authorities_db` now?

Reviewer: I don't have an answer for this but wanted to make sure we do not land this change with a TODO in it (@patrickkuo ?)

Reviewer 2: client_db folder is created by the wallet when it's first started

Author: Do you mean when we run this?
wallet --no-shell new-address

Because simply starting wallet doesn't do it.

Author: Also, that command assumes the files lives in my home dir:

```
Error: Cannot open wallet config file at "/Users/claymurphy/.sui/sui_config/wallet.conf"
```

So I have to run:

```
claymurphy@Clays-MacBook-Pro tmp % wallet --config wallet.conf --no-shell new-address
```

@awelc how is this easier than having someone manually `mkdir` the directory of their choosing for all of this?

Reviewer: I was not aware of this. I would suggest having `client_db`'s default location to be `~/.sui/sui_config/` as well exactly for the the reason @Clay-Mysten mentions - we don't want to force a specific directory creation, but we may want to allow it. We could also make the directory for wallet to be `~/.sui/wallet_config` and have both `wallet.conf`, `wallet.key` and `client_db` to go there by default. @patrickkuo?

Reviewer 2: the wallet will write the `client_db` to the same dir as the `wallet.conf` , it will cause problem if user starts 2 instances of wallet connecting to 2 different network if we hardcode the client_db to `~/.sui/wallet_config`

Author: @patrickkuo But in my case, I ran the `wallet new-address` command in my `tmp' directory where `wallet.conf` lives, specifying it with the `--config` flag, only to have a path of:

```
tmp/tmp/client_db
```

That doesn't seem right.

Author: See my input and output above.

The network configuration is stored in `network.conf` and can be used
subsequently to start the network. The `wallet.conf` and `wallet.key`
are also created to be used by the Sui wallet to manage the newly
created accounts.

> By default, these files are placed in your home directory at

Reviewer: ```suggestion
By default, these files are placed in the
```
It's obvious from the path that it's home directory plus "directory" was used twice.

Author: I removed the second "directory" here. Thanks for the catch.

> `~/.sui/sui_config` directory. But you can override this location by

Reviewer: ```suggestion
`~/.sui/sui_config` directory (created automatically if it does not yet exist). But you can override this location by
```

Just to make it crystal clear that the default directory is created for you whereas the custom one is not (the latter being already explicit).

providing an alternative path to the `--working-dir` argument. Run
the command like so to place the files in the `dir` directory:

```shell
$ sui genesis --working-dir /path/to/sui/config/dir
```

:note: That path and directory must already exist and will not be
created with the `--working-dir` argument.

### Recreating Genesis

To recreate Sui genesis state in the same location, which will remove
existing configuration files, pass the `--force` option to the `sui

> genesis` command and either run it in the same directory or specify

Reviewer: ```suggestion
genesis` command and either run it in the default directory (`~/.sui/sui_config`) or specify
```

it once again, using the `--working-dir` argument:

```shell
$ sui genesis --force --working-dir /path/to/sui/config/dir
```

For example:

```shell

> $ sui genesis --force --working-dir /Users/name/tmp

Reviewer: I don't think that this example adds any value on top of the one immediately preceding it.

Author: @patrickkuo what do you think?

Author: Removing...

:warning: **Do not use in production**: Keys are stored in file!

## Starting the network
Run the following command to start the local Sui network, assuming you
accepted the default location for configuration:

```shell
$ sui start
```

This command will look for the Sui network configuration file
`network.conf` in the `~/.sui/sui_config` directory. But you can
override this setting by providing a path to the directory where
this file is stored:

```shell
$ sui start --config /path/to/sui/network/config/file
```

For example:

```shell
$ sui start --config /Users/name/tmp/network.conf
```

TODO: Find out why I get this error:

```
claymurphy@Clays-MacBook-Pro tmp % sui start --config /Users/claymurphy/tmp/network.conf
Error: Address already in use (os error 48)
```

I see no Sui-specific processes running:

```
claymurphy@Clays-MacBook-Pro tmp % ps -ef | grep -i sui
  501   442     1   0 12:02PM ??         0:00.22 /System/Library/CoreServices/CoreServicesUIAgent.app/Contents/MacOS/CoreServicesUIAgent
  501   623     1   0 12:03PM ??         0:01.10 /System/Library/PrivateFrameworks/AppleMediaServicesUI.framework/amsengagementd

>   501  1433     1   0  1:01PM ??         0:00.21 /System/Library/PrivateFrameworks/PaperKit.framework/Contents/LinkedNotesUIService.app/Contents/MacOS/LinkedNotesUIService

Reviewer: I am fishing here - REST API server running locally?

Reviewer 2: you can check which command is using the port using `lsof -i tcp:10000`

    `transfer`       Transfer an object
Use `help <command>` to see more information on each command.

You can start the wallet in two modes: interactive shell or command line interface.

### Interactive shell

To start the interactive shell, execute the following (in a different

> terminal window than one used to execute `sui start`). Assuming you

Reviewer: This is now somewhat redundant with the first paragraph of the "Using the wallet" section

one coin to split, one for the gas payment.

Let us examine objects owned by address `45CDA12E3BAFE3017B4B3CD62C493E5FBAAD7FB0`:

```shell
$ wallet --no-shell objects --address 45CDA12E3BAFE3017B4B3CD62C493E5FBAAD7FB0
```

> With output:

Reviewer: ```suggestion
With output resembling:
```
